packages/nextjs-mf/utils/createDynamicFederatedPage.ts

```diff
@@ -0,0 +1,120 @@
+import * as React from 'react'
+import { importDelegatedModule } from '@module-federation/utilities'
+import type { RuntimeRemote, WebpackRemoteContainer } from '@module-federation/utilities'
+import { useRouter } from 'next/router'
+import { GetServerSideProps, Redirect } from 'next/types'
+
+export type PathResolver = (path: string) => {
+  remoteContainer: string | RuntimeRemote,
+  modulePath: string
+}
+
+function defaultPathResolver(path: string) {
+  const trimmedPath = path.replace(/^\/+/, '')
+  const slashIndex = trimmedPath.indexOf('/')
+  const remoteContainer = slashIndex > -1 ? trimmedPath.substring(0, slashIndex) : trimmedPath
+  const modulePath = slashIndex > -1 ? trimmedPath.substring(slashIndex) : '/index'
+
+  return {
+    remoteContainer,
+    modulePath: `.${modulePath}`
+  }
+}
+
+export type ErrorHandler = (error: unknown) => { redirect: Redirect } | { notFound: true }
+
+function defaultErrorHandler(error: unknown) {
+  console.error(error)
+  return {
+    notFound: true,
+  } as const
+}
+
+function removeQueryParams(path: string) {
```

Comment:
parsing path with `new URL()`, but don't know if it is better than `path.split('?')[0]`

Reply:
new url bit more reliable, as long as its valid url not a partial like /endpoint? Thing